### ISSUE TITLE
Unexpected memory use /crash on simple filters from parquet backend

### ISSUE THREAD

**Issue by cboettig** [duckdb/duckdb#2193]
**What does happen?**
I see unexpected memory use that frequently crashes me out when doing some basic filters on a VIEW of a large parquet file.  

**What should happen?**
duckdb should use only as much RAM as set by `PRAGMA memory_limit` instead of crashing.

**To Reproduce**

Apologies I cannot reproduce this with a more minimal example!  Below is an example with a 150 GB parquet file I'm using, including download link.  I'm happy to try something different on my end too to help debug.

```r

library(duckdb)
library(DBI)
library(dplyr)

# download data
file <- "part-0.parquet.parquet"
download.file("https://minio.cirrus.carlboettiger.info/shared-data/birddb/parquet/part-0.parquet", file)

conn <- DBI::dbConnect(duckdb::duckdb())
# Ceate a View so we can see the table via DBI
DBI::dbExecute(conn, paste0("PRAGMA memory_limit='", 4, "GB'"))
query <- paste0("CREATE VIEW '", "ebd",
                "' AS SELECT * FROM parquet_scan('",
                file, "');")
DBI::dbSendQuery(conn, query)
## attempt operation on table

bench::bench_time({
out <- tbl(conn,"ebd") %>% filter(COUNTRY=="Mexico", `COMMON NAME`=="Wood Thrush") %>% collect()
})

out 
```
This query shown should only return about 8000 rows of data, but executing this uses just over 150 GB (!) of RAM (or on a smaller machine, creates an OOM error crashing the session).  (Using current GitHub version of duckdb).  

I attempted to reproduce this in SQL cli but my SQL is apparently a little rusty.  In the cli, it's not necessary to create the View, but we seem to need this to use parquet from R over the DBI interface(?).  

(Aside: I attempted to use `duckdb_register_arrow` strategy here instead, but running that reliably crashes me out even on my 200 GB platform.  However, I suspect that's due to a separate bug in arrow 5.0, because I get a similar out-of-memory behavior if I try the same filter in pure `arrow`, which [their team suspects is connected to a known bug](https://issues.apache.org/jira/browse/ARROW-13761?focusedCommentId=17405511&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17405511)

**Environment (please complete the following information):**
 - OS: Linux
 - DuckDB Version (  Current GitHub release, R version `0.2.9`, not sure how to get the R client to tell me the duckdb version)

**Before submitting**
- [ x] Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?
- [ x] Have you tried this on the latest `master` branch? In case you cannot compile, you may find some binaries here: https://github.com/duckdb/duckdb/releases/tag/master-builds

**Comment by hannesmuehleisen** [comment]
Thanks, we will look into this. Obviously this should not happen.

**Comment by hannesmuehleisen** [comment]
For the CWI crowd, file is downloading at jewels07:/export/scratch2/home/hannes/bug2193 , still some time to go.

**Comment by Mytherin** [comment]
This should be fixed in #2201. Could you verify that that fixes the problem for you?

**Comment by cboettig** [comment]
Thanks @Mytherin , that was fast!  Yup, looks good on my end now too!
